numpy/lib/_arrayterator_impl.pyi

```diff
@@ -7,10 +7,9 @@ from typing import (
 )
 
 from numpy import ndarray, dtype, generic
-from numpy._typing import DTypeLike, NDArray
+from numpy._typing import DTypeLike, NDArray, _Shape as _AnyShape
 
-# TODO: Set a shape bound once we've got proper shape support
-_Shape = TypeVar("_Shape", bound=Any)
+_Shape = TypeVar("_Shape", bound=_AnyShape)
```

Comment:
Isn't this typevar usually called `_ShapeType`?
Could add a `# TODO: rename to _ShapeType` for a future PR.

Reply:
Yes that makes sense 👌🏻 I didn't want to rename it in this PR